compiler/circle-resizer/include/CircleModel.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __CIRCLE_RESIZER_CIRCLE_MODEL_H__
+#define __CIRCLE_RESIZER_CIRCLE_MODEL_H__
+
+#include "Shape.h"
+
+#include <string>
+#include <memory>
+#include <vector>
+
+namespace luci
+{
+class Module;
+}
```

Comment:
Why not  `#include <luci/IR/Module.h>` ?

Reply:
In the current state a user of `circle_resizer_core` does not requires any other dependencies (`luci_lang` in this case). For sure we can still have in cmake:
```
target_link_libraries(circle_resizer_core PUBLIC luci_lang)
```
For me such encapsulation is more clear but from practical benefits I can now only indicate reduction area of re-compilation after some change in `luci/IR/Module.h` - so it's not a very big argument

But in general my opinion here is not very strong so I can change it ;-)

Reply:
Thanks for explanation. The current code looks okay.